cylc/uiserver/jupyter_config.py

```diff
@@ -57,8 +57,64 @@
     )
 ]
 
-# store the JupyterHub runtime files in ~/.cylc/hub
+# store JupyterHub runtime files in the user config directory
 USER_CONF_ROOT.mkdir(parents=True, exist_ok=True)
 c.JupyterHub.cookie_secret_file = f'{USER_CONF_ROOT / "cookie_secret"}'
 c.JupyterHub.db_url = f'{USER_CONF_ROOT / "jupyterhub.sqlite"}'
 c.ConfigurableHTTPProxy.pid_file = f'{USER_CONF_ROOT / "jupyterhub-proxy.pid"}'
+
+# write Cylc logging to the user config directory
+# NOTE: Parallel UIS instances will conflict over this file.
+#       See https://github.com/cylc/cylc-uiserver/issues/240
+c.CylcUIServer.logging_config = {
+    'version': 1,
+    'handlers': {
+        'file': {
+            'class': 'logging.handlers.RotatingFileHandler',
+            'level': 'INFO',
+            'filename': str(USER_CONF_ROOT / 'uiserver.log'),
+            'mode': 'a',
+            'backupCount': 5,
+            'maxBytes': 10485760,
```

Comment:
Copied from Cylc, perhaps we should reduce this a little?

Reply:
Given the amount of logging the UIS doesn't produce I feel that this is likely to be overkill.